### PR TITLE
first draft of QDK tutorial template

### PR DESCRIPTION
**Pull request by Bradben**
Markdown template for QDK tutorials

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Lowest<br>category<br>score | Spelling<br>issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/tutorial-template.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/07d1639827b9ba1fe03f64b5fc7d368bcc3a03cb/articles/tutorial-template.md) | 100 | 100 | 0 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/e1aaf7ee-fa42-4c22-a264-cb49ff51d9cd_report.html) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: articles/tutorial-template.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+# Tutorial: <verb and action> 
+ <!-- (example) "Tutorial: Write and Simulate Qubit-level programs in Q#" -->
+
+ <!-- 
+    * Provide a high-level abstract of the tutorial: what the user will do, accomplish, and learn. "In this tutorial you will do..." 
+
+    * Provide a bullet list of what is covered, based on the H2 procedure steps 
+-->
+
+ ## Prerequisites
```
Comment (cgranade) [inline articles/tutorial-template.md:38]:
```suggestion
## Prerequisites
```

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Lowest<br>category<br>score | Spelling<br>issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/tutorial-template.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/537f762adad2e4e6d0b971a8c2096106151b532b/articles/tutorial-template.md) | 100 | 100 | 0 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/1dde514a-ca98-482a-838c-47f2e1e4333b_report.html) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: articles/tutorial-template.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
+title: Markdown template for quantum tutorials
```
Comment (natke) [inline articles/tutorial-template.md:2]: Use the h1 text here too (this is the meta title that google uses).

File: articles/tutorial-template.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+description: Template and guidance for writing tutorials for the Microsoft Quantum Development Kit documentation.
```
Comment (natke) [inline articles/tutorial-template.md:3]: Make it clear that this meta description used for SEO, and that this is what is displayed on a search results page

Comment (natke) [reply articles/tutorial-template.md:3]: If you put the intent text in here, then we don't need to worry about the "As a content developer ..." stuff below.

File: articles/tutorial-template.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+-->
+
+# Tutorial: <verb and action> 
+ <!-- (example) "Tutorial: Write and Simulate Qubit-level programs in Q#" -->
+
+ <!-- 
+    * Provide a high-level abstract of the tutorial: what the user will do, accomplish, and learn. "In this tutorial you will do..." 
+
+    * Provide a bullet list of what is covered, based on the H2 procedure steps
```
Comment (natke) [inline articles/tutorial-template.md:35]: Instead of a comment here, can we have the bulleted list?

Comment (natke) [reply articles/tutorial-template.md:35]: You can add the formatting too:

```
> [!div class="checklist"]
> * Create and connect to a VM
> * Select and use VM images
> * View and use specific VM sizes
> * Resize a VM
> * View and understand VM state
```

**Comment by natke** [comment]
Oh, the location of this file should be in the root of the repository (one level up), otherwise it will be published to the live site.

**Comment by geduardo** [comment]
I like it. One suggestion, maybe we should specify some guidelines on how to handle the multiple host programs. For example, all the host code (Procedure 2) should be at least (when possible) in: Python, C# VSCode, C# VS
This could be relevant also for Procedure 3, however, sometimes the interpretation tools differ between platforms (e.g. it comes to my mind the postprocessing and visualization in the QML sample, we have the code in Python but not in C#)

**Comment by cgranade** [comment]
> One suggestion, maybe we should specify some guidelines on how to handle the multiple host programs. For example, all the host code (Procedure 2) should be at least (when possible) in: Python, C# VSCode, C# VS

https://github.com/microsoft/iqsharp/issues/24 has in some cases limited what Python samples we can provide. In particular, when a sample is split into multiple C# projects (e.g.: the simulation samples), Python is not supported as a host.

File: tutorial-template.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
+---
+title: Markdown template for quantum tutorials
+description: Template and guidance for writing tutorials for the Microsoft Quantum Development Kit documentation. 
+author: bradben
+ms.author: bradben
+ms.date: 04/01/2020
+ms.topic: article
+uid: microsoft.quantum.tutorial-template
+
+# title: <title should match the H1 title - Google displays this on the search results page. (example) "Tutorial: Write and Simulate Qubit-level programs in Q#">
```
Comment (cjgronlund) [inline tutorial-template.md:10]: Title need only match H1 in intent. Most important: Intent displays in search result without truncating key terms. Also, make sure titles and headings are sentence case.

title: <title states article intent succinctly so it fits on the Google search results page. Follow guidance in https://aka.ms/seo-for-writers-cheat-sheet. Ex.: "Tutorial: Write & simulate qubit-level programs in Q#

File: tutorial-template.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        * Include links to other content (such as our doc set or APIs) in the procedures. Explain everything that the user needs to know to do the task. This helps keep them focused and on task. EXCEPTION for the Prerequisites section for links to installation, configuration, etc.
+        * Include a 'More Info', 'Resources', or 'See Also' section
+    * AVOID:
+        * Notes, Tips, and Important boxes. Users find them distracting and generally skip over them. 
+        * Excessive bulleted or numbered lists. Use only as needed for clarity. 
+
+-->
+
+# Tutorial: <verb and action> 
+ <!-- (example) "Tutorial: Write and Simulate Qubit-level programs in Q#". This should match the title in the metadata-->
```
Comment (cjgronlund) [inline tutorial-template.md:37]: Sentence-case headings: Tutorial: Write and simulate qubit-level programs in Q#

File: tutorial-template.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    * DO NOT: 
+        * Include links to other content (such as our doc set or APIs) in the procedures. Explain everything that the user needs to know to do the task. This helps keep them focused and on task. EXCEPTION for the Prerequisites section for links to installation, configuration, etc.
+        * Include a 'More Info', 'Resources', or 'See Also' section
+    * AVOID:
+        * Notes, Tips, and Important boxes. Users find them distracting and generally skip over them. 
+        * Excessive bulleted or numbered lists. Use only as needed for clarity. 
+
+-->
+
+# Tutorial: <verb and action>
```
Comment (cjgronlund) [inline tutorial-template.md:36]:
```suggestion
# Tutorial: <State what the tutorial accomplishes> 
```
"verb and action" seems redundant.

File: tutorial-template.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+## Procedure_3 (Results)
+<!-- "This procedure analyzes the output of the scenario..."  -->
+
+<!-- Display the results and explain how they relate to the code and concept 
+
+     Create output results for each of the host code examples used in Procedure 2.
+
+-->
+
+## Procedure_4 (Variations and exercises)
```
Comment (cjgronlund) [inline tutorial-template.md:105]: Remove variations and exercises section. A tutorial -- as our doc group defines - show one prescriptive way to accomplish a scenario.

File: tutorial-template.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+## Procedure_1 (Build the Scenario)
+<!-- "This procedure walks you through writing the code to..."  -->
+
+<!-- Include:
+    * A short sentence or two describing what happens in this section
+    * List and link the steps performed in this section
+    * [optional] Display the finished and commented Q# code block. It can be beneficial to see the big picture first. 
+-->
+
+### Step_1
```
Comment (cjgronlund) [inline tutorial-template.md:67]: I recommend keeping the heading hierarchy flat. Makes the individual parts of the task easier to find. We've seen customers search on procedures that were just one part of an article.
```suggestion
## Describe step 1
```

File: tutorial-template.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+### Step_1 
+<!-- Step names should describe an action, for example, "Add the Quantum.Operations namespace definition", "Measure the qubits and store the results">
+<!-- Include
+    * Brief description of what this step does
+    * Individual steps as needed
+    * Explain what this step accomplishes, what commands were used
+    * Other pertinent notes or comments
+-->
+### Step_2
```
Comment (cjgronlund) [inline tutorial-template.md:75]:
```suggestion
## Describe step 2
```

File: tutorial-template.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+### Step_3
```
Comment (cjgronlund) [inline tutorial-template.md:77]:
```suggestion
## Describe step 3
```

File: tutorial-template.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+
+## Procedure_2 (Run the Scenario)
+<!-- "This procedure walks you through running the code to..."  -->
+
+<!-- Include:
+    * A short sentence or two describing what happens in this section
+    * (If more than 2 or 3 steps) List and link the steps performed in this section
+    * A tabbed structure with code examples for the Python (if supported for the scenario), C# for VS, and C# for VS Code hosts'
+-->
+### Step_1
```
Comment (cjgronlund) [inline tutorial-template.md:87]: Not sure if this level of complexity is needed - sections of steps. Let's chat about it.

File: tutorial-template.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+    * Other pertinent notes or comments
+-->
+## Summary
+<!-- Review the scenario and highlight:
+    * Key learning takeaways
+    * Other possible applications of this knowledge
+    * New concepts and commands introduced
+-->
+
+## Next Steps
```
Comment (cjgronlund) [inline tutorial-template.md:121]:
```suggestion
## Next steps
```
Sentence case headings

File: tutorial-template.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+
+## Prerequisites
+ <!-- 
+ Provide as necessary:
+     * Software versions of installations
+     * Previous tutorials or knowledge
+     * Preparations such as creating a host project (this can be a link to How to Create a Q# Project)
+-->
+
+## Procedure_1 (Build the Scenario)
```
Comment (cjgronlund) [inline tutorial-template.md:58]:
```suggestion
## Procedure_1: Write Q# code
```

File: tutorial-template.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+    * Brief description of what this step does
+    * Individual steps as needed
+    * Explain what this step accomplishes, what commands were used
+    * Other pertinent notes or comments
+-->
+### Step_2
+
+### Step_3
+
+## Procedure_2 (Run the Scenario)
```
Comment (cjgronlund) [inline tutorial-template.md:79]:
```suggestion
## Procedure_2: Write the host code (Run the scenario)
```

File: tutorial-template.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+### Step_1
+<!-- Include (as needed)
+    * Brief description of what this step does
+    * Individual steps as needed
+    * Explain what this step accomplishes, what commands were used
+    * Other pertinent notes or comments
+-->
+### Step_2
+
+## Procedure_3 (Results)
```
Comment (cjgronlund) [inline tutorial-template.md:96]:
```suggestion
## Procedure_3: Analyze the results
```

File: tutorial-template.md (hunk 8 above)
Comment (cjgronlund) [reply tutorial-template.md:105]: Variations could be a tutorial doc listed under "Next steps"

**Pull request merged**